src/moonlink/src/table_handler.rs

```diff
@@ -587,6 +457,189 @@ impl TableHandler {
+                    && lsn >= *force_snapshot_lsns.iter().next().as_ref().unwrap().0
+                    && !*mooncake_snapshot_ongoing;
+
+                match xact_id {
+                    Some(xact_id) => {
+                        // Discard streaming write buffer, if content already persisted.
+                        if Self::to_discard(lsn, initial_persistence_lsn) {
+                            table.abort_in_stream_batch(xact_id);
+                            return;
+                        }
+                        if let Err(e) = table.commit_transaction_stream(xact_id, lsn).await {
+                            error!(error = %e, "stream commit flush failed");
+                        }
+                    }
+                    None => {
+                        if Self::to_discard(lsn, initial_persistence_lsn) {
+                            return;
+                        }
+                        table.commit(lsn);
+                        if table.should_flush() || force_snapshot {
+                            if let Err(e) = table.flush(lsn).await {
+                                error!(error = %e, "flush failed in commit");
+                            }
+                        }
+                    }
+                }
+
+                if force_snapshot {
+                    Self::reset_iceberg_state_at_mooncake_snapshot(
+                        iceberg_snapshot_result_consumed,
+                        iceberg_snapshot_ongoing,
+                    );
+                    assert!(table.create_snapshot(SnapshotOption {
+                        force_create: true,
+                        skip_iceberg_snapshot: *iceberg_snapshot_ongoing,
+                        skip_file_indices_merge: *maintainance_ongoing,
+                        skip_data_file_compaction: *maintainance_ongoing,
+                    }));
+                    *mooncake_snapshot_ongoing = true;
+                }
+            }
+            TableEvent::StreamAbort { xact_id } => {
+                table.abort_in_stream_batch(xact_id);
+            }
+            TableEvent::Flush { lsn } => {
+                if Self::to_discard(lsn, initial_persistence_lsn) {
+                    return;
+                }
+                if table.is_in_initial_copy() {
+                    table
+                        .initial_copy_buffered_events
+                        .push(TableEvent::Flush { lsn });
+                    return;
+                }
+                if let Err(e) = table.flush(lsn).await {
+                    error!(error = %e, "explicit flush failed");
+                }
+            }
+            TableEvent::StreamFlush { xact_id } => {
+                if let Err(e) = table.flush_transaction_stream(xact_id).await {
+                    error!(error = %e, "stream flush failed");
+                }
+            }
+            _ => {}
+        }
+    }
+
+    fn to_discard(lsn: u64, initial_persistence_lsn: Option<u64>) -> bool {
```

Comment:
I think next time when refactor it's good not to remove the comment...
I don't think the `to_discard` naming is clear enough alone...

Reply:
In the context of usage, it's clear; but reading the function alone is not, that's why I added a comment.
I think it's the responsibility for refactor-er to rename or leave comment?